### PR TITLE
Add recovery control health check type

### PR DESCRIPTION
**Pull request by ankita10051991**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

 Closes [#20730](https://github.com/hashicorp/terraform-provider-aws/issues/20730)

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSRoute53HealthCheck_withRoutingControlArn'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSRoute53HealthCheck_withRoutingControlArn -timeout 180m
=== RUN   TestAccAWSRoute53HealthCheck_withRoutingControlArn
=== PAUSE TestAccAWSRoute53HealthCheck_withRoutingControlArn
=== CONT  TestAccAWSRoute53HealthCheck_withRoutingControlArn
--- PASS: TestAccAWSRoute53HealthCheck_withRoutingControlArn (94.69s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	104.767s

...
```

### CODE REVIEW

File: .changelog/20731.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+```release-note:enhancement
+resoucre_aws_route53_health_check: Add `RECOVERY_CONTROL` health check type and `routing_control_arn` attriute
```
Comment (anGie44) [inline .changelog/20731.txt:3]: Small edits 😃 
```suggestion
resource_aws_route53_health_check: Add `RECOVERY_CONTROL` health check type and `routing_control_arn` argument
```

Comment (ankita10051991) [reply .changelog/20731.txt]: Fixed :)

**Comment by AdamTylerLynch** [comment]
@anGie44 we are seeing the linter flag on the docs, even though the value is valid. Any ideas here?

```
ERROR: File "./website/docs/r/route53_health_check.html.markdown", block #5 (lines 84-98):
1 issue(s) found:

Error: "RECOVERY_CONTROL" is an invalid value as type (aws_route53_health_check_invalid_type)

  on /var/folders/h8/p5dkjq2d7gv9wky4m1bz80_h12rnp_/T/tmp.LGBhK74i/main.tf line 11:
  11:   type                = "RECOVERY_CONTROL"
```
https://github.com/hashicorp/terraform-provider-aws/runs/3475661838?check_suite_focus=true#step:8:62

File: website/docs/r/route53_health_check.html.markdown [hunk 2]
```diff
@@ -110,6 +128,7 @@ The following arguments are supported:
 * `cloudwatch_alarm_region` - (Optional) The CloudWatchRegion that the CloudWatch alarm was created in.
 * `insufficient_data_health_status` - (Optional) The status of the health check when CloudWatch has insufficient data about the state of associated alarm. Valid values are `Healthy` , `Unhealthy` and `LastKnownStatus`.
 * `regions` - (Optional) A list of AWS regions that you want Amazon Route 53 health checkers to check the specified endpoint from.
+* `routing_control_arn` - (Optional) The Amazon Resource Name (ARN) for the Route 53 Application Recovery Controller routing control. This is used when health check type is RECOVERY_CONTROL
```
Comment (anGie44) [inline website/docs/r/route53_health_check.html.markdown:131]:
```suggestion
* `routing_control_arn` - (Optional) The Amazon Resource Name (ARN) for the Route 53 Application Recovery Controller routing control. This is used when health check type is `RECOVERY_CONTROL`.
```

File: aws/resource_aws_route53_health_check.go [hunk 3]
```diff
@@ -155,6 +155,12 @@ func resourceAwsRoute53HealthCheck() *schema.Resource {
 				Default:  false,
 			},
 
+			"routing_control_arn": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
```
Comment (anGie44) [inline aws/resource_aws_route53_health_check.go:161]:
```suggestion
				Optional: true,
				ValidateFunc: validateArn,
```

File: website/docs/r/route53_health_check.html.markdown (hunk 2 above)
Comment (ankita10051991) [reply website/docs/r/route53_health_check.html.markdown]: fixed

File: aws/resource_aws_route53_health_check_test.go [hunk 4]
```diff
@@ -690,3 +715,19 @@ resource "aws_route53_health_check" "test" {
...
+resource "aws_route53recoverycontrolconfig_routing_control" "test" {
+  name        = "tf-acc-test"
+  cluster_arn = aws_route53recoverycontrolconfig_cluster.test.arn
+}
+
+resource "aws_route53_health_check" "test" {
+  type                = "RECOVERY_CONTROL"
+  routing_control_arn = aws_route53recoverycontrolconfig_routing_control.test.arn
+}
+`
```
Comment (anGie44) [inline aws/resource_aws_route53_health_check_test.go:733]: you'll also want to randomly generate these names e.g.
```suggestion
func testAccRoute53HealthCheckConfigRoutingControlArn (rName string) string {
    return fmt.Sprintf(`
resource "aws_route53recoverycontrolconfig_cluster" "test" {
  name = %[1]q
}

resource "aws_route53recoverycontrolconfig_routing_control" "test" {
  name        = %[1]q
  cluster_arn = aws_route53recoverycontrolconfig_cluster.test.arn
}

resource "aws_route53_health_check" "test" {
  type                = "RECOVERY_CONTROL"
  routing_control_arn = aws_route53recoverycontrolconfig_routing_control.test.arn
}
`, rName)
}
```

File: aws/resource_aws_route53_health_check_test.go [hunk 5]
```diff
@@ -333,6 +333,31 @@ func TestAccAWSRoute53HealthCheck_Disabled(t *testing.T) {
...
+	var check route53.HealthCheck
+	resourceName := "aws_route53_health_check.test"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		ErrorCheck:   testAccErrorCheck(t, route53.EndpointsID),
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckRoute53HealthCheckDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRoute53HealthCheckConfigRoutingControlArn,
```
Comment (anGie44) [inline aws/resource_aws_route53_health_check_test.go:346]:
```suggestion
				Config: testAccRoute53HealthCheckConfigRoutingControlArn(rName),
```

File: aws/resource_aws_route53_health_check_test.go (hunk 4 above)
Comment (ankita10051991) [reply aws/resource_aws_route53_health_check_test.go]: done

File: aws/resource_aws_route53_health_check_test.go [hunk 6]
```diff
@@ -333,6 +333,31 @@ func TestAccAWSRoute53HealthCheck_Disabled(t *testing.T) {
 	})
 }
 
+func TestAccAWSRoute53HealthCheck_withRoutingControlArn(t *testing.T) {
+	var check route53.HealthCheck
+	resourceName := "aws_route53_health_check.test"
```
[the remark this answers is not in the document]
Comment (ankita10051991) [reply aws/resource_aws_route53_health_check_test.go:339]: done

File: aws/resource_aws_route53_health_check_test.go (hunk 5 above)
Comment (ankita10051991) [reply aws/resource_aws_route53_health_check_test.go]: done

**Review by anGie44 — APPROVED** [review]
LGTM 🚀 thanks again @ankita10051991 

Output of acceptance test:
```
--- PASS: TestAccAWSRoute53HealthCheck_withRoutingControlArn (66.16s)
```

**Comment by anGie44** [comment]
Noting CI failure has been addressed with: https://github.com/hashicorp/terraform-provider-aws/pull/20741

**Pull request merged**